batch/batch/batch.py

```diff
@@ -107,6 +108,7 @@ def batch_record_to_dict(app, record):
             if rv['delta_cores_mcpu'] != 0 and instance.state == 'active':
                 instance.adjust_free_cores_in_memory(rv['delta_cores_mcpu'])
                 scheduler_state_changed.set()
+                cancel_ready_state_changed.set()
```

Comment:
If the last parent of a job completes, the child will transition from pending to ready.  If the child is cancelled (either because the batch is cancelled or that or another parent failed), it will be ready for cancellation from the ready state.